invokeai/app/invocations/latent.py

```diff
@@ -363,9 +360,78 @@ def invoke(self, context: InvocationContext) -> ImageOutput:
                 session_id=context.graph_execution_state_id, node=self
             )
 
+            torch.cuda.empty_cache()
+
             context.services.images.save(image_type, image_name, image, metadata)
             return build_image_output(
-                image_type=image_type,
-                image_name=image_name,
-                image=image
+                image_type=image_type, image_name=image_name, image=image
             )
+
+
+LATENTS_INTERPOLATION_MODE = Literal[
+    "nearest", "linear", "bilinear", "bicubic", "trilinear", "area", "nearest-exact"
+]
+
+
+class ResizeLatentsInvocation(BaseInvocation):
+    """Resizes latents to explicit width/height."""
+
+    type: Literal["lresize"] = "lresize"
+
+    # Inputs
+    latents: Optional[LatentsField]             = Field(description="The latents to resize")
+    width: int                                  = Field(ge=64, multiple_of=8, description="The width to resize to (px)")
+    height: int                                 = Field(ge=64, multiple_of=8, description="The height to resize to (px)")
+    downsample: Optional[int]                   = Field(default=8, ge=1, description="The downsampling factor (leave at 8 for SD)")
+    mode: Optional[LATENTS_INTERPOLATION_MODE]  = Field(default="bilinear", description="The interpolation mode")
+    antialias: Optional[bool]                   = Field(default=False, description="Whether or not to antialias (applied in bilinear and bicubic modes only)")
+
+    def invoke(self, context: InvocationContext) -> LatentsOutput:
+        latents = context.services.latents.get(self.latents.latents_name)
+
+        resized_latents = torch.nn.functional.interpolate(
+            latents,
+            size=(
+                self.height // self.downsample,
+                self.width // self.downsample,
+            ),
+            mode=self.mode,
+            antialias=self.antialias if self.mode in ["bilinear", "bicubic"] else False,
+        )
+
+        # https://discuss.huggingface.co/t/memory-usage-by-later-pipeline-stages/23699
+        torch.cuda.empty_cache()
+
+        name = f"{context.graph_execution_state_id}__{self.id}"
+        context.services.latents.set(name, resized_latents)
+        return LatentsOutput(latents=LatentsField(latents_name=name))
+
+
+class ScaleLatentsInvocation(BaseInvocation):
+    """Scales latents by a given factor."""
+
+    type: Literal["lscale"] = "lscale"
+
+    # Inputs
+    latents: Optional[LatentsField]             = Field(description="The latents to scale")
+    scale_factor: float                         = Field(ge=0, description="The factor by which to scale the latents")
```

Comment:
This probably needs to be `gt 0` and not `ge 0`.

Reply:
Another thing I'm conflicted about. Valid vs useful. Ok, I'm make it gt.